Mirror/Runtime/NetworkTransformBase.cs

```diff
@@ -0,0 +1,545 @@
+﻿// vis2k:
+// base class for NetworkTransform and NetworkTransformChild.
+// New method is simple and stupid. No more 1500 lines of code.
+//
+// Server sends current data.
+// Client saves it and interpolates last and latest data points.
+//   Update handles transform movement / rotation
+//   FixedUpdate handles rigidbody movement / rotation
+//
+// Notes:
+// * Built-in Teleport detection in case of lags / teleport / obstacles
+// * Quaternion > EulerAngles because gimbal lock and Quaternion.Slerp
+// * Syncs XYZ. Works 3D and 2D. Saving 4 bytes isn't worth 1000 lines of code.
+// * Initial delay might happen if server sends packet immediately after moving
+//   just 1cm, hence we move 1cm and then wait 100ms for next packet
+// * Only way for smooth movement is to use a fixed movement speed during
+//   interpolation. interpolation over time is never that good.
+//
+using System;
+using UnityEngine;
+
+namespace Mirror
+{
+    public abstract class NetworkTransformBase : NetworkBehaviour
+    {
+        [Range(0, 1)] public float sendInterval = 0.1f; // every ... seconds
+
+        // rotation compression. not public so that other scripts can't modify
+        // it at runtime. alternatively we could send 1 extra byte for the mode
+        // each time so clients know how to decompress, but the whole point was
+        // to save bandwidth in the first place.
+        // -> can still be modified in the Inspector while the game is running,
+        //    but would cause errors immediately and be pretty obvious.
+        [Tooltip("Compresses 16 Byte Quaternion into None=12, Some=6, Much=3, Lots=2 Byte")]
+        [SerializeField] Compression compressRotation = Compression.Much;
+        public enum Compression { None, Some, Much, Lots }; // easily understandable and funny
+
+        // server
+        Vector3 lastPosition;
+        Quaternion lastRotation;
+
+        // client
+        public class DataPoint
+        {
+            public float timeStamp;
```

Comment:
should be double

Reply:
timeStamp is set to Unity's Time.time, it's not sent over the network and it doesn't use NetworkTime.time either